### PR TITLE
[new-app] Write PropTypes for each components

### PR DESCRIPTION
**Pull request by jbaek7023**
PropTypes makes the code more readable and testable to developers.

### CODE REVIEW

File: src/javascript/app_2/components/elements/popover.jsx [hunk 1]
```diff
@@ -10,7 +11,7 @@ class Popover extends React.Component {
     }
 
     render() {
-        const popver = (
+        const popover = (
```
Comment (jbaek7023) [inline src/javascript/app_2/components/elements/popover.jsx]: This looks like a typo here.

File: src/javascript/app_2/components/elements/popover.jsx [hunk 2]
```diff
@@ -10,7 +11,7 @@ class Popover extends React.Component {
     }
 
     render() {
-        const popver = (
```
Comment (jbaek7023) [inline src/javascript/app_2/components/elements/popover.jsx]: This looks like a typo here.

File: src/javascript/app_2/components/elements/data_table.jsx [hunk 3]
```diff
@@ -82,4 +83,14 @@ class DataTable extends React.Component {
     }
 }
 
+DataTable.propTypes = {
+    columns         : PropTypes.array,
+    data_source     : PropTypes.array,
+    is_full_width   : PropTypes.bool,
```
Comment (morteza-binary) [inline src/javascript/app_2/components/elements/data_table.jsx]: @jbaek7023 
it's better to define properties in alphabetical order.

Comment (jbaek7023) [reply src/javascript/app_2/components/elements/data_table.jsx]: Ok! working on it now

Comment (morteza-binary) [reply src/javascript/app_2/components/elements/data_table.jsx]: 👍

File: src/javascript/app_2/components/layout/header.jsx [hunk 4]
```diff
@@ -63,10 +64,11 @@ const DrawerFooter = () => ( // TODO: update the UI
 );
 
 class Header extends React.Component {
+
     render() {
         return (
             <React.Fragment>
-                <header id={this.props.id} className='shadow'>
```
Comment (jbaek7023) [inline src/javascript/app_2/components/layout/header.jsx]: 'id' props is never used.

File: src/templates/.eslintrc.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+    rules: {
+        'react/prop-types': 0,
```
Comment (binary-hamedanchi) [inline src/templates/.eslintrc.js]: In order to enable this rule for other folders, please remove it from `/.eslintrc.js`.

File: package.json [hunk 6]
```diff
@@ -111,6 +111,7 @@
     "mobx-utils": "3.2.2",
     "moment": "2.19.4",
     "promise-polyfill": "6.1.0",
+    "prop-types": "^15.6.1",
```
Comment (binary-hamedanchi) [inline package.json]: Please use fixed versioning. (`"15.6.1"`)

**Pull request closed without merging**